pipe_anchorages/transforms/create_in_out_events.py

```diff
@@ -159,6 +160,13 @@ def _create_in_out_events(self, records):
 
             last_timestamp = rcd.timestamp
             last_state = state
+        if (
+            last_state in self.in_port_states
+            and rcd.is_possible_gap_end
```

Comment:
@bitsofbits I've lost track of how exactly `is_possible_gap_end` works, but do you think this condition is required here?

Reply:
Nope. It is normally applied to the next record and rcd is the previous record. I guess I should have checked your answer before I went and made this exact fix myself :-/